docs/programming-guide.md

```diff
@@ -1086,6 +1086,62 @@ for details.
 </tr>
 </table>
 
+### Shuffle operations
+
+Certain operations within Spark trigger an event known as the shuffle. The shuffle is Spark's 
+mechanism for re-distributing data so that is grouped differently across partitions. This typically 
+involves re-arranging and copying data across executors and machines, making shuffle a complex and 
```

Comment:
re-arranging and copying are redundant.

Also, be consistent on "shuffle" vs. "the shuffle".